### PR TITLE
Gatk V3.5

### PR DESCRIPTION
**Pull request by jerowe**
* [x] I have read the guidelines above.
* [x] This PR adds a new recipe.
* [x] This PR updates an existing recipe.

Hi Johannes,

This PR is to update the version of GATK (gatk-framework) from 3.4.x to 3.5. It also updates the bin script.

It follows the same structure as the 3.4.x release. Credit really goes to this guy for the build system. https://github.com/chapmanb/gatk . ;-)

Best,
Jillian

### CODE REVIEW

File: recipes/gatk-framework/gatk-framework [hunk 1]
```diff
@@ -6,17 +6,7 @@
...
-SOURCE="${BASH_SOURCE[0]}"
-while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symlink
-    DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
-    SOURCE="$(readlink "$SOURCE")"
-    [[ $SOURCE != /* ]] && SOURCE="$DIR/$SOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
-done
-DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
-
-JAR_DIR=$DIR
+JAR_DIR="$(cd "$(dirname "$0")" && pwd -P)"
```
Comment (johanneskoester) [inline recipes/gatk-framework/gatk-framework]: I don't immediately get this. Can you explain why this can replace the code on the left?

**Comment by johanneskoester** [comment]
Thanks @jerowe!
@chapmanb, this is your baby. Can you have a look?

**Comment by chapmanb** [comment]
Jillian;
Thanks for doing this. It looks like your code package gatk-protected, so should be a different recipe. The original gatk-framework is only the MIT licensed parts of the code (I can bump that part to 3.5 if it is useful to folks) so does not include variant calling and other license restricted code.

For gatk-protected, it's not clear we're allowed to redistribute it due to the terms of the licensing. See the discussion in #1247 for current thoughts. Thanks for helping move this forward.

**Comment by jerowe** [comment]
Hi @chapmanb and @johanneskoester 

Thanks for putting in the work to make the uberjar in the first place! I'm not a java person, and I never would have figured that out. If I'd known you were still activate I would have asked you for help. ;-)

I was a bit unclear on the licensing bit myself, but I thought it would check out as ok since the software is free for non profit use. My thoughts were the responsibility was on the user once the licensing was clear. I also put a sentence when you start up the gatk-framework stating as such. I took a look at the discussion. 

I see there was talking of having a register script. I'd be happy to work on that if no one else can. We need GATK 3.5 for an inhouse pipeline, and its getting that up and running is a priority for me right now.

Best,
Jillian

**Comment by jerowe** [comment]
Taking a closer look at the licensing is making me a bit paranoid, so I'm going to delete the repo I forked from gatk-protected.

**Comment by jerowe** [comment]
There is an email in the licensing info. I'm going to send off a quick email and ask.

**Comment by jerowe** [comment]
@chapmanb , if you wanted to update your GATK to 3.5 that would be awesome.

btw, the more I thought about it the more I realized I should have just asked you first. I'm really sorry if I stepped on your toes! I get tunnel vision and don't think sometimes. ;-)

**Comment by chapmanb** [comment]
Jillian -- thanks much, I appreciate you getting things started with moving to GATK 3.5. I pushed a new recipe built off GATK 3.5 now, so hopefully this will work for your analyses. Thanks again.

**Comment by jerowe** [comment]
@chapman, thanks for the new release! ;-)

**Pull request closed without merging**